### DISCUSSION TITLE
Customized error messages with validation details based on JOI Schema

### DISCUSSION THREAD

**Discussion by bartosz-szymanski** [react-hook-form/react-hook-form#1814] [Q&A] [answered]
Hi! 👋🏻

I'm trying to customize error messages in my reusable component containing Material UI Field wrapped into Controller and I met some obstacles. Let's talk about min length error for example. 
I'm using currently newest RC **6.0.0-rc.3** and JOI Schema to validate:
` name: Joi.string().trim().min(3).required()`
And my reusable Material UI Text Field Wrap looks in following way:

```
 <Controller
      as={(
        <TextField
          error={!!error}
          helperText={error && t(`form.error.${error.type}`)}
          label={t(`${translationKey}.${name}.value`)}
          variant="outlined"
        />
      )}
      name={name}
      control={control}
      defaultValue={defaultValue}
    />
```
I would like to pass an extra argument to translations with min expected length to make a clear and obvious message to the user with all details. 
The error looks like:
```
{
  message: ""name" length must be at least 3 characters long", 
  type: "string.min"
}
```
So I can easily identify error type by property **type** and it is great. But if I would like to get **value** I see 3 possibilities and all of them are not satisfying:

- use RegExp to get it somehow from property **message** - it is awful
- use ValidationSchema, but I don't want to pass it to every wrapped form field - it is redundant
- use implicit value as another parameter to my wrapped component - it is redundant duplication

If anyone has already figured it out please let me know. A simple message like 'Field value is too short' does not meet my requirements.

Best regards,
Bartosz

**Comment by bluebill1049** [comment, marked as answer]
perhaps you want to use `resolver` to customize the errors object return?

**Reply by bartosz-szymanski** [reply]
Could you provide more details or an example?

**Reply by bluebill1049** [reply]
Take a look at here: https://react-hook-form.com/api#validationResolver

**Reply by toto1384** [reply]
The link is not working anymore.
